sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -103,6 +106,13 @@ object DataType {
 
   def fromJson(json: String): DataType = parseDataType(parse(json))
 
+  // Until Spark-2.1, we use json strings for defining schemas in user-facing APIs.
+  // Since we add an user-friendly API in the DDL parser, we employ DDL formats for the case.
+  // To keep back-compatibility, we use `fromJson` first, and then try the new API.
+  def fromString(text: String): DataType = {
```

Comment:
I mean users should use `fromString` just for DDL formats.

It is pretty weird that `from_json` is used to parse both json and DDL format.

How about add a new `from_ddl` or `from_string`?